### PR TITLE
feat: mf-5567 add friend tech plugin

### PR DESCRIPTION
**Pull request by UncleBill**
https://mask.atlassian.net/browse/MF-5567

### CODE REVIEW

**Comment by UncleBill** [comment]
@guanbinrui PTAL

File: packages/plugins/FriendTech/src/base.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    description: { fallback: PLUGIN_DESCRIPTION },
+    publisher: { name: { fallback: '' }, link: '' },
+    enableRequirement: {
+        supports: {
+            type: 'opt-out',
+            sites: {
+                [EnhanceableSite.Localhost]: true,
+            },
+        },
+        target: 'beta',
```
Comment (guanbinrui) [inline packages/plugins/FriendTech/src/base.ts]:
```suggestion
        target: 'stable',
```

File: packages/plugins/FriendTech/src/SiteAdaptor/components/HoldingList.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            return FriendTech.getHolding(address, nextIndicator)
+        },
+        getNextPageParam: (x) => x.nextIndicator,
+    })
+    const holdings = useMemo(() => {
+        if (!data?.pages.length) return EMPTY_LIST
+        // There could be duplicate users.
+        return uniqBy(
+            data.pages.flatMap((x) => x.data),
+            (x) => x.address,
```
Comment (guanbinrui) [inline packages/plugins/FriendTech/src/SiteAdaptor/components/HoldingList.tsx:57]:
```suggestion
            (x) => x.address.toString(),
```

Comment (UncleBill) [reply packages/plugins/FriendTech/src/SiteAdaptor/components/HoldingList.tsx:57]: It's already `string`

Comment (UncleBill) [reply packages/plugins/FriendTech/src/SiteAdaptor/components/HoldingList.tsx:57]: Or you mean `.toLowerCase()`? Actually no need, data are consistent from FriendTech's API.

**Review by guanbinrui — DISMISSED** [review]

**Pull request merged**